department-of-veterans-affairs/va.gov-cms
# VACMS-16853: next build branches ftw

## Description

closes #16853

## Testing done

- Updated some tests for git information/API/frontend. 
- Manually tested locally by running script.
- Need to test on Tugboat with this PR.

## Screenshots

Content release form when you can trigger a build.
<img width="889" alt="Screenshot 2024-02-08 at 1 43 57 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/3640707/9910d980-6bf7-41f4-973d-6f254c67437b">

Content release form when build is running with a link to the build log.
<img width="816" alt="Screenshot 2024-02-08 at 1 46 02 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/3640707/47acf8d3-23c4-46a6-9630-3c1d40b38837">

## QA steps

1. Log in and go to "/admin/content/deploy/next"
   - [x] Confirm you can select versions of next-build and vets-website from git references
   - [x] Confirm there is no build in progress
   - [x] No request or lock files should exist
   - [x] The versions for next-build and vets-website should be "__default"
2. Submit the form with two branches chosen for next-build and vets-website
   - [x] Confirm that a build request file exists. 
   - [x] No build should have started yet so no lock file should exist.
   - [x] The versions for next-build and vets-website should be reported as what you selected.
2. Refresh the page a few times until the background process kicks off from reading the build request file.
   - [x] Confirm that the form fields are locked during the build.
   - [x] Confirm you can view the build log file and it is adding data until the build finishes.
4. Refresh the form page after the build is complete
   - [x] Confirm the disabled fields are no longer disabled.
   - [x] View the log to see that the build finished. It might have an error, but that will be addressed in future tickets.

### Definition of Done


### Select Team for PR review
